### PR TITLE
Dev add s3

### PR DESCRIPTION
**Pull request by zhangrc**
add object storage support for s3 protocol

### CODE REVIEW

**Comment by zhangrc** [comment]
#297 , add s3 support ,thanks for check

File: scripts/animatediff.py [hunk 1]
```diff
@@ -161,7 +161,60 @@ def on_ui_settings():
...
+            "enable to Store file in object storage that supports the s3 protocol",
+            gr.Checkbox,
+            section=section
+        )
+    )
+    shared.opts.add_option(
+        "animatediff_s3_host",
+        shared.OptionInfo(
+            None,
+            "s3 protocol host",
```
Comment (continue-revolution) [inline scripts/animatediff.py:177]: capital letter at the beginning

File: scripts/animatediff_output.py [hunk 2]
```diff
@@ -12,6 +12,9 @@
 
 from scripts.animatediff_logger import logger_animatediff as logger
 from scripts.animatediff_ui import AnimateDiffProcess
+import boto3
+from botocore.exceptions import ClientError
```
Comment (continue-revolution) [inline scripts/animatediff_output.py:16]: import on demand

File: scripts/animatediff_output.py [hunk 3]
```diff
@@ -274,7 +277,7 @@ def _save(
                     file.container_metadata["Title"] = infotext
                     file.container_metadata["Comment"] = infotext
                 file.write(video_array, codec="vp9", fps=params.fps)
-
+        [self._save_to_s3_stroge(video_path) for video_path in video_paths]
```
Comment (continue-revolution) [inline scripts/animatediff_output.py:280]: do not use list. this writing style is weird.

File: scripts/animatediff_output.py [hunk 4]
```diff
@@ -302,3 +305,48 @@ def _encode_video_to_b64(self, paths):
...
+    def _save_to_s3_stroge(self ,file_path):
+        """
+        put object to object storge
+        :type bucketname: string
+        :param bucketname: will save to this 'bucket' , access_key and secret_key must have permissions to save 
+        :type file  : file 
+        :param file : the local file 
+        """        
+        s3_enable =shared.opts.data.get("animatediff_s3_enable", False) 
+        if not s3_enable:
```
Comment (continue-revolution) [inline scripts/animatediff_output.py:328]: move this check to `_save`

**Comment by zhangrc** [comment]
I have modified it, please check

**Pull request merged**